authors/apps/authentication/serializers.py

```diff
@@ -57,17 +103,27 @@ def validate(self, data):
                 'A password is required to log in.'
             )
 
+        """
+        Check email address and password at a time so that a user
+        can know exactly what went wrong
+        """
+        email_ = User.objects.filter(email=email)
+        if not email_.exists():
+            raise serializers.ValidationError(
+                'A user with email address "{}" does not exist'.format(email)
+            )
         # The `authenticate` method is provided by Django and handles checking
         # for a user that matches this email/password combination. Notice how
         # we pass `email` as the `username` value. Remember that, in our User
         # model, we set `USERNAME_FIELD` as `email`.
+
         user = authenticate(username=email, password=password)
 
         # If no user was found matching this email/password combination then
         # `authenticate` will return `None`. Raise an exception in this case.
         if user is None:
             raise serializers.ValidationError(
-                'A user with this email and password was not found.'
+                'You typed a wrong password, please try again'
```

Comment:
Any reasons why you changed this message? Seems better as it is to me.

Reply:
This message wasn't descriptive enough to me, at first, a user could not know which of the two between email or password that were typed is wrong. So I refactored the code so that I specifically notify the user, which of the two wasn't typed correctly. First I first check the email alone to find out if the user exists, then I proceed to authenticate to check whether the user typed the password correctly. There I can clearly notify the user what specifically went wrong

Reply:
I think it's better to let this message be generic (leave as is). Otherwise, this can allow user enumeration.

Reply:
alright